examples/platforms/nrf528xx/nrf52840/README.md

```diff
@@ -126,15 +126,107 @@ You can prefix the compiler command using the CCPREFIX parameter. This speeds up
+By default, OpenThread uses mbedTLS library with support for CryptoCell 310 hardware acceleration of cryptographic operations. You can disable CryptoCell 310 and use software cryptography instead by building OpenThread with the following parameter:
 ```
-$ make -f examples/Makefile-nrf52840 DISABLE_CC310=1
+$ make -f examples/Makefile-nrf52840 DISABLE_BUILTIN_MBEDTLS=0 ...
 ```
 
-### Optional mbedTLS threading support
-By default, mbedTLS library is built without support for multiple threads. You can enable this built-in support by building OpenThread with the following parameter:
+When building external application with OpenThread libraries and software cryptography please use:
+- Crypto libraries:
+  - `output/nrf52840/lib/libmbedcrypto.a` (present after successful build)
+- Include directories:
+  - `third_party/NordicSemiconductor/libraries/nrf_security/mbedtls_platform_config`
+  - `third_party/mbedtls`
+  - `third_party/mbedtls/repo/include`
+- Defines:
+  - `MBEDTLS_CONFIG_FILE` set to  `"mbedtls-config.h"`
+  - `MBEDTLS_USER_CONFIG_FILE` set to `"nrf52840-mbedtls-config.h"`
+
+### CryptoCell 310 abort mechanism
+
+CryptoCell 310 platform implements abort mechanism. By default any fault in CryptoCell 310 library results in system reset. This behaviour can be changed by a call to following function which MUST be called before `mbedtls_platform_setup`:
+```
+void nrf_cc310_platform_set_abort(
+	nrf_cc310_platform_abort_apis_t const * const apis)
+```
+
+### CryptoCell 310 RTOS support
+
+Hardware accelerated mbedTLS library supports access from multiple threads. By default simple blocking implementation of mutexes is used. Used locking mechanism can be replaced using following function which MUST be called before `mbedtls_platform_setup`:
+```
+void nrf_cc310_platform_set_mutexes(nrf_cc310_platform_mutex_apis_t const * const apis,
+                                    nrf_cc310_platform_mutexes_t const * const mutexes)
+```
+
+Sample implementation of mutexes for FreeRTOS and Zephyr operating systems can be found in `nrf_cc310_platform_mutex_freertos.c` and `nrf_cc310_platform_mutex_zephyr.c`. Both files implements function `int nrf_cc310_platform_mutex_init(void)` which MUST be called before `mbedtls_platform_setup`.
+
+Sample implementation of abort functionality for FreeRTOS and Zephyr operating systems can be found in `nrf_cc310_platform_abort_freertos.c` and `nrf_cc310_platform_abort_zephyr.c`. Both files implements function `void nrf_cc310_platform_abort_init(void)` which MUST be called before `mbedtls_platform_setup`.
+
+Typical setup in RTOS environment (error checking omitted):
+```
+mbedtls_platform_set_calloc_free(calloc, free);
+nrf_cc310_platform_abort_init();
+nrf_cc310_platform_mutex_init();
+mbedtls_platform_setup(NULL);
+```
+
+When building external application that uses RTOS with OpenThread libraries and CryptoCell 310 hardware acceleration please use:
```

Comment:
```suggestion
When building an external application that uses RTOS with OpenThread libraries and the CryptoCell 310 hardware acceleration, use the following configuration:
```
Shouldn't this be in a separate section?

Reply:
Well, it's in CC310 RTOS section and tells the configuration that you should use only for CC310 + RTOS so I think that this section is ok.